scripts/sct_fmri_moco.py

```diff
@@ -12,81 +12,112 @@
-                      example=['nn', 'linear', 'spline'])
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description="""Remove temporary files.""",
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value="multiple_choice",
-                      description="""Verbose.""",
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1', '2'])
+    # parser initialisation
+    parser = argparse.ArgumentParser(
+        description=("Motion correction of fMRI data. Some robust features include:\n"
+                     "    - group-wise (-g)\n"
+                     "    - slice-wise regularized along z using polynomial function (-p)\n"
+                     "      (For more info about the method, type: isct_antsSliceRegularizedRegistration)\n"
+                     "    - masking (-m)\n"
+                     "    - iterative averaging of target volume\n"
+                     "\n"
+                     "The outputs of the motion correction process are:\n"
+                     "    - the motion-corrected fMRI volumes\n"
+                     "    - the time average of the corrected fMRI volumes\n"
+                     "    - a time-series with 1 voxel in the XY plane, for the X and Y motion direction (two separate "
+                     "files), as required for FSL analysis.\n"
+                     "    - a TSV file with the slice-wise average of the motion correction for XY (one file), that "
+                     "can be used for Quality Control.\n"),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Input data (4D). Example: fmri.nii.gz"
+    )
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-g',
+        metavar=Metavar.int,
+        type=int,
+        help="Group nvols successive fMRI volumes for more robustness."
+    )
+    optional.add_argument(
+        '-m',
+        metavar=Metavar.file,
+        help="Binary mask to limit voxels considered by the registration metric."
+    )
+    optional.add_argument(
+        '-param',
+        metavar=Metavar.list,
+        type=list_type(',', str),
+        help=(f"R|Advanced parameters. Assign value with \"=\"; Separate arguments with \",\".\n"
+              f"    -poly [int]: Degree of polynomial function used for regularization along Z. For no regularization "
```

Comment:
same cosmetic suggestion wrt. displaying lists
```suggestion
              f"  - poly [int]: Degree of polynomial function used for regularization along Z. For no regularization "
```